ibis/expr/analysis.py

```diff
@@ -804,75 +736,29 @@ def predicate(expr):
         return None
 
 
-def fully_originate_from(exprs, parents):
-    def finder(expr):
-        op = expr.op()
+def _find_root_table(expr):
+    op = expr.op()
+
+    if isinstance(op, ops.Selection):
+        return lin.proceed, op.table
```

Comment:
I'm not sure I can fully reason about this condition, but my guess is that it omits the topmost selection.